src/components/EventManagement/UpdateEvent.jsx

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from "react";
+import { getEventDataID, eventUpdateData, getEventdata } from "../../api";
+import { useParams, useNavigate } from "react-router-dom";
+
+const AddNewEvents = () => {
+  const [eventData, setEventData] = useState({});
+
+  const params = useParams();
+  const { id } = params;
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const fetchSingalData = () => {
+      try {
+        getEventDataID(id).then((res) => {
+          setEventData(res);
+        });
+      } catch (err) {
+        console.log(err);
+      }
+    };
+    fetchSingalData();
```

Comment:
no need to create a new function `fetchSingalData` for fetching the event data. Please keep the logic in useEffect and remove the `fetchSingalData` function.

Reply:
done